ext/mysqli/mysqli.stub.php

```diff
@@ -0,0 +1,773 @@
+<?php
+
+class mysqli_driver
+{
+}
+
+class mysqli
+{
+    /**
+     * @param ?string $host
+     * @param ?string $user
+     * @param ?string $password
+     * @param ?string $database
+     * @param ?int $port
+     * @param ?string $socket
+     */
+    public function __construct(?string $host = null, ?string $user = null, ?string $password = null, ?string $database = null, ?int $port = null, ?string $socket = null);
+
+    /**
+     * @param ?bool $mode
+     *
+     * @return bool
+     */
+    public function autocommit(?bool $mode);
+
+    /**
+     * @param ?int $flags
+     * @param ?string $name
+     *
+     * @return bool
+     */
+	public function begin_transaction(?int $flags = 0, ?string $name = '');
+
+	/**
+	 * @param ?string $user
+	 * @param ?string $password
+	 * @param ?string $database
+	 *
+	 * @return bool
+	 */
+	public function change_user(?string $user, ?string $password, ?string $database);
+
+	/**
+	 * @return string
+	 */
+	public function character_set_name();
+
+    /**
+     * @return bool
+     */
+	public function close();
+
+	/**
+	 * @param ?int $flags
+	 * @param ?string $name
+	 *
+	 * @return bool
+	 */
+	public function commit(?int $flags = 0, ?string $name = '');
```

Comment:
these are not nullable parameters (there is no `!` in the ZPP). That said, the `$name` parameter's default value should be `UNKNOWN` AFAIK.

Reply:
how can I know that it should be UNKNOWN?

Reply:
There is some information written about it by Nikita: https://externals.io/message/106522 (the whole thread is interesting, but you should search for "UNKNOWN"). :)